core/src/main/web/app/mainapp/components/notebook/codecell-directive.js

```diff
@@ -70,22 +91,24 @@
           if ($scope.isLocked()) {
             return false;
           }
-          if ($scope.cellmodel.input.hidden === true) {
-            return false;
-          }
-          return true;
+          return $scope.cellmodel.input.hidden !== true;
         };
 
         $scope.bkNotebook = getBkNotebookWidget();
-        // ensure cm refreshes when 'unhide'
-        $scope.$watch('isShowInput()', function(newValue, oldValue) {
-          if ($scope.cm && newValue === true && newValue !== oldValue) {
+
+        //ensure cm refreshes when 'unhide'
+        $scope.afterShow = function () {
+          $scope.cm.refresh();
+        };
+        $scope.$watch('cellmodel.input.hidden', function(newValue, oldValue) {
+          if ($scope.cm && oldValue === true && newValue !== oldValue) {
             bkUtils.fcall(function() {
-              $scope.cm.refresh();
+              $scope.afterShow();
```

Comment:
Do we need the call to afterShow here if the ngHideEx directive handles that automatically?